File: server/fitness_api/schemas/diary.py
from typing import Optional
from datetime import date as _date
from .config import BaseModelWithConfig
from pydantic import Field
from uuid import UUID


class BaseDiaryModel(BaseModelWithConfig):
    user_id: UUID = Field(..., title="User ID", alias="userId")
    date: _date = Field(..., title="the journaling date")


class PostDiaryModel(BaseDiaryModel):
    ...


class PatchDiaryModel(BaseModelWithConfig):
    maximum_calorie_intake: Optional[int] = Field(
        None,
        title="The maximum amount of calories that a user can take in per day",
        alias="maximumCalorieIntake",
    )
    total_calorie_intake: Optional[int] = Field(
        None,
        title="Calories that the user has taken in during the day",
        alias="totalCaloriesIntake",
    )


class DiaryModel(BaseDiaryModel):
    id: UUID = Field(..., title="Diary ID")
    breakfast_id: UUID = Field(..., alias="breakfastId", title="one of the meals")
    lunch_id: UUID = Field(..., alias="lunchId", title="one of the meals")
    dining_id: UUID = Field(..., alias="diningId", title="one of the meals")
    maximum_calorie_intake: int = Field(
        ...,
        title="The maximum amount of calories that a user can take in per day",
        alias="maximumCalorieIntake",
    )
    total_calorie_intake: int = Field(
        ...,
        title="Calories that the user has taken in during the day",
        alias="totalCaloriesIntake",
    )

File: lib/screen/diary/components/diary.dart
class Diary {
  late String id;
  late String userId;
  late String date;
  late String breakfastId;
  late String lunchId;
  late String diningId;
  late int maximumCaloriesIntake;
  late int totalCaloriesIntake;

  Diary(this.id, this.userId, this.date, this.breakfastId, this.lunchId,
      this.diningId, this.maximumCaloriesIntake, this.totalCaloriesIntake);

  Diary.fromJson(Map<String, dynamic> json) {
    id = json["id"];
    userId = json["userId"];
    date = json["date"];
    breakfastId = json["breakfastId"];
    lunchId = json["lunchId"];
    diningId = json["diningId"];
    maximumCaloriesIntake = json["maximumCaloriesIntake"];
    totalCaloriesIntake = json["totalCaloriesIntake"];
  }
}
